src/main/ngapp/src/app/dataservices/shared/view.model.ts

```diff
@@ -105,6 +104,75 @@ export class View {
     return "SELECT * FROM " + connectionName.toLowerCase() + VdbsConstants.SCHEMA_MODEL_SUFFIX + "." + sourceNodeName + ";";
   }
 
+  /**
+   * Duplicate sources are not added.
+   *
+   * @param {SchemaNode} sourceToAdd the source being added
+   */
+  public addSource( sourceToAdd: SchemaNode ): void {
+    const index = this.sources.findIndex( ( source ) => source.getName() === sourceToAdd.getName() );
+
+    if ( index === -1 ) {
+      this.sources.push( sourceToAdd );
+    }
+  }
+
```

Comment:
getName() is not adequate for identifying the source, but I think you are aware of that - and we can change to path with subsequent commit